src/otx/cli/cli.py

```diff
@@ -0,0 +1,337 @@
+            scheduler_kwargs = {"instantiate": False, "fail_untyped": False, "skip": {"optimizer"}}
+            if enable_auto_config:
+                # Add Default values from Auto-Configurator
+                if "--optimizer" not in sys.argv:
+                    optim_kwargs["default"] = auto_configurator.load_default_optimizer_config()
+                if "--scheduler" not in sys.argv:
+                    scheduler_kwargs["default"] = auto_configurator.load_default_scheduler_config()
+            sub_parser.add_subclass_arguments(
+                baseclass=(Optimizer,),
+                nested_key="optimizer",
+                **optim_kwargs,
+            )
+            sub_parser.add_subclass_arguments(
+                baseclass=LRSchedulerTypeTuple,
+                nested_key="scheduler",
+                **scheduler_kwargs,
+            )
+
+            skip: set[str | int] = set(self.engine_subcommands()[subcommand])
+            fn = getattr(Engine, subcommand)
+            description = get_short_docstring(fn)
+            added = sub_parser.add_method_arguments(
+                Engine,
+                subcommand,
+                skip=skip,
+                fail_untyped=False,
+            )
+            if data_root is not None and subcommand == "train":
+                # Add Default values of Callbacks & Logger
+                default_engine_config = auto_configurator.load_default_engine_config()
+                default_engine_config = flatten_dict(default_engine_config)
+                sub_parser.set_defaults(default_engine_config)
+            if data_root is not None:
+                sub_parser.set_defaults(
+                    **{
+                        "data.config.data_root": data_root,
+                        "engine.data_root": data_root,
+                    },
+                )
+
+            if "logger" in added:
+                sub_parser.link_arguments("engine.work_dir", "logger.init_args.save_dir")
+
+            self._subcommand_method_arguments[subcommand] = added
+            self._subcommand_parsers[subcommand] = sub_parser
+            parser_subcommands.add_subcommand(subcommand, sub_parser, help=description)
+
+    def _set_extension_subcommands_parser(self, parser_subcommands: _ActionSubCommands) -> None:
+        from otx.cli.install import add_install_parser
+
+        add_install_parser(parser_subcommands)
+
+    def instantiate_classes(self) -> None:
+        """Instantiate the necessary classes based on the subcommand.
+
+        This method checks if the subcommand is one of the engine subcommands.
+        If it is, it instantiates the necessary classes such as config, datamodule, model, and engine.
+        """
+        if self.subcommand in self.engine_subcommands():
+            self.config_init = self.parser.instantiate_classes(self.config)
+            self.datamodule = self._get(self.config_init, "data")
+            self.model, optimizer, scheduler = self.instantiate_model()
```

Comment:
They can be considered partially dependent. But the reason I've separated them here is because the optimizer and scheduler should be partially instantiated (the optimizer depends on model.params, and the scheduler depends on the optimizer). Do you have any recommendations for a different way to do this?